ceremonies/src/lib.rs

```diff
@@ -627,8 +628,8 @@ impl<T: Trait> Module<T> {
         let perdegree = day / T::Moment::from(360u32);
         let start = next - duration;
         // rounding to the lower integer degree. Max error: 240s = 4min
-        let abs_lon: i32 = mlocation.lon.abs().lossy_into();
-        let abs_lon_time = T::Moment::from(abs_lon as u32) * perdegree;
+        let abs_lon: u32 = i64::lossy_from(mlocation.lon.abs()).saturated_into();
```

Comment:
lon is signed! If unit tests pass, then we need better test coverage!

Reply:
But we are taking the absolute value here?

Reply:
true that